### ISSUE TITLE
PETScWrappers::PreconditionEisenstat crashes

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#2969]
See TODO comment in tests/petsc/reinit_preconditioner_02.cc from #2963:

```
petsc/reinit_preconditioner_02.debug: RUN failed. ------ Additional output on stdout/stderr:

[0]PETSC ERROR: ------------------------------------------------------------------------
[0]PETSC ERROR: Caught signal number 11 SEGV: Segmentation Violation, probably memory access out of range
[0]PETSC ERROR: Try option -start_in_debugger or -on_error_attach_debugger
[0]PETSC ERROR: or see http://www.mcs.anl.gov/petsc/documentation/faq.html#valgrind
[0]PETSC ERROR: or try http://valgrind.org on GNU/linux and Apple Mac OS X to find memory corruption errors
[0]PETSC ERROR: likely location of problem given in stack below
[0]PETSC ERROR: ---------------------  Stack Frames ------------------------------------
[0]PETSC ERROR: Note: The EXACT line numbers in the stack are not available,
[0]PETSC ERROR:       INSTEAD the line number of the start of the function
[0]PETSC ERROR:       is given.
[0]PETSC ERROR: [0] MatMultDiagonalBlock_SeqAIJ_Inode line 4002 /ssd/candi/deal.II-toolchain-build/petsc-3.6.3/src/mat/impls/aij/seq/inode.c
[0]PETSC ERROR: [0] MatMultDiagonalBlock line 2159 /ssd/candi/deal.II-toolchain-build/petsc-3.6.3/src/mat/interface/matrix.c
[0]PETSC ERROR: [0] PCApply_Eisenstat line 41 /ssd/candi/deal.II-toolchain-build/petsc-3.6.3/src/ksp/pc/impls/eisens/eisen.c
[0]PETSC ERROR: [0] PCApply line 463 /ssd/candi/deal.II-toolchain-build/petsc-3.6.3/src/ksp/pc/interface/precon.c
```

**Comment by tamiko** [comment]
@tjhei Is this resolved with #2963 ?

**Comment by tjhei** [comment]
Nope.
